### ISSUE TITLE
diff_pool - strange assignment matrix

### ISSUE THREAD

**Issue by jnsLs** [rusty1s/pytorch_geometric#595]
## 🐛 Bug

Dear all,
I am trying to reproduce the results shown in _Z. Ying, J. You, C. Morris, X. Ren, W. Hamilton, and J. Leskovec,“Hierarchical  graph  representation  learning  with  differentiable pooling,”  inAdvances  in  Neural  Information  Processing  Systems,2018, pp. 4801–4811._ for the enzymes dataset using the provided script "enzymes_diff_pool.py". 

In the paper, a method is suggested to visualize the clusters of atoms after pooling. It allows for assigning atoms to their respective cluster by finding the maximum value of each row in the assignment matrix.

Utilizing this approach, I visualize the clusters after pooling for both the original code of the paper and the pytorch module. While for the original code, the atoms are assigned to various clusters, for the pytorch script all atoms are assigned to, at the most, 2 clusters. 

## To Reproduce

I am using the default setting of both codes. I only added the visualization.
Those are the last view training epochs using pytorch geometric: 
Epoch: 143, Train Loss: 0.1162907, Val Acc: 0.4500000, Test Acc: 0.3500000
Epoch: 144, Train Loss: 0.1151333, Val Acc: 0.4666667, Test Acc: 0.3500000
Epoch: 145, Train Loss: 0.1142788, Val Acc: 0.4666667, Test Acc: 0.3500000
Epoch: 146, Train Loss: 0.1130905, Val Acc: 0.4666667, Test Acc: 0.3500000
Epoch: 147, Train Loss: 0.1142356, Val Acc: 0.4333333, Test Acc: 0.3500000
Epoch: 148, Train Loss: 0.1456007, Val Acc: 0.4166667, Test Acc: 0.3500000
Epoch: 149, Train Loss: 0.2427479, Val Acc: 0.4000000, Test Acc: 0.3500000
Epoch: 150, Train Loss: 1.2312759, Val Acc: 0.3333333, Test Acc: 0.3500000

## Expected behavior

I would expect more clusters or rather an evolving clustering, with increasing model accuracy. Instead, it seems that the assignment matrix is not improving.

## Environment

* OS: Linux (Kubuntu)
* Python version: 3.7
* PyTorch version: 1.1.0
* CUDA/cuDNN version: 10.1
* GCC version: 8.3.0
* Any other relevant information:

**Comment by rusty1s** [comment]
I see, the example outputs a rather low accuracy which could explain your observation. I wonder what has caused this. Accuracy should be much higher. I will look into it.

**Comment by jiachangliu** [comment]
Hello @jnsLs I have also encountered this problem. Are you able fix it?
